Source/AIExtensions/Private/StateTree/Tasks/AIExtStateTreeTaskSendGameplayEvent.cpp

```diff
@@ -0,0 +1,49 @@
+#include "StateTree/Tasks/AIExtStateTreeTaskSendGameplayEvent.h"
+
+#include "AbilitySystemComponent.h"
+
+#include <StateTreeExecutionContext.h>
+
+FAIExtStateTreeTaskSendGameplayEvent::FAIExtStateTreeTaskSendGameplayEvent()
+{
+    // No tick needed.
+    bShouldCallTick = false;
+    // No need to update bound properties after enter state.
+    bShouldCopyBoundPropertiesOnTick = false;
+    bShouldCopyBoundPropertiesOnExitState = false;
+}
+
+EStateTreeRunStatus FAIExtStateTreeTaskSendGameplayEvent::EnterState( FStateTreeExecutionContext & context, const FStateTreeTransitionResult & transition ) const
+{
+    TRACE_CPUPROFILER_EVENT_SCOPE_STR( __FUNCTION__ );
+
+    const auto & instance_data = context.GetInstanceData< FInstanceDataType >( *this );
+
+    auto asc = instance_data.TargetASC;
+
+    if ( asc == nullptr )
+    {
+        asc = instance_data.Target ? instance_data.Target->FindComponentByClass< UAbilitySystemComponent >() : nullptr;
```

Comment:
suggestion to use `UAbilitySystemGlobals::GetAbilitySystemComponentFromActor` instead
a bit cleaner + will try to find the asc using the interface before trying to find by component

Reply:
done in other branch